docs/overview/community/ComponentGuidelines.stories.mdx

```diff
@@ -0,0 +1,178 @@
+import { Title, Meta } from "@storybook/addon-docs";
+
+<Meta title="Overview/Community/Component Guidelines" />
+
+<Title>Component Guidelines</Title>
+
+This guide aims to provide insights and recommendations for building and contributing
+robust and adaptable components that are consistent with the existing UI Kit library.
+
+## File Structure & Purpose
+
+The directory should contain the following files:
+
+```sh
+MyComp/
+├── index.ts            # entry file with the component's public exports
+├── MyComp.tsx          # component implementation and types
+├── MyComp.stories.tsx  # component documentation and examples
+├── MyComp.styles.ts    # component classes and stying utils
+└── MyComp.test.tsx     # component unit tests
+```
+
+## Guidelines
+
+- Components and their types MUST be named in PascalCase and prefixed with `Hv`.
+- Components MUST be exported from the `index.ts` file.
+- Boolean props MUST be opt-in (default is `false`).
+- Event handlers MUST be named `on<Action>`.
+
+### Styles
+
+- Use `createClasses` utility to declare classes.
+- Class names in `createClasses` are exposed to users, so be mindful of naming:
+  - `classes.root` MUST be on the root element.
+  - Make it clear what element the class targetting when it isn't the root element.
+  - If conditional, the class name should be based on the condition.
+- Leverage `theme` to style component. Don't hard-code colors or spacings
+
+### Documentation
+
+- Add a `Component.stories.tsx` exporting multiple [CSF](https://storybook.js.org/docs/react/api/csf) stories
+- The primary story SHOULD complete be named `Main` and have configured [controls](https://storybook.js.org/docs/react/essentials/controls)
+- Avoid adding too many similar stories; each story should have a unique purpose
+
+### Others
+
+- Use `@testing-library/react` for unit tests.
+- Prefer semantic HTML elements instead of custom logic whenever possible
+- Follow [WCAG](https://www.w3.org/TR/WCAG21/) and [Using ARIA](https://www.w3.org/TR/using-aria/) guidelines
+
+## Example
+
+Here's an example anatomy of a `MyComp` component:
+
+```ts
+// MyComp.styles.ts
+
+export const { staticClasses, useClasses } = createClasses("MyComp", {
+  /** Applied to the root element */
+  root: {
+    // leverage the 👇 `theme.` object
+    padding: theme.spacing("xs"),
+    backgroundColor: theme.colors.backgroundColor,
+  },
+  /** Applied to the root element when selected */
+  selected: {},
+  /** Applied to the root element when disabled */
+  disabled: {
+    // 👇 leverage global `disabled` instead of adding a `buttonDisabled`
+    "& $button": {
+      cursor: "not-allowed",
+    },
+  },
+  /** Applied to the button element */
+  button: {},
+});
+```
+
+```tsx
```

Comment:
Do you think we should add the export for the MyComp classes type and static classes: `MyCompClasses` and `myCompClasses`? We do this in every component.

Reply:
yeah, makes sense 👍